state/processList.go

```diff
@@ -1166,9 +1166,15 @@ func NewProcessList(state interfaces.IState, previous *ProcessList, dbheight uin
 		pl.AdminBlock = adminBlock.NewAdminBlock(previous.AdminBlock)
 		pl.EntryCreditBlock, err = entryCreditBlock.NextECBlock(previous.EntryCreditBlock)
 	} else {
-		pl.DirectoryBlock = directoryBlock.NewDirectoryBlock(nil)
-		pl.AdminBlock = adminBlock.NewAdminBlock(nil)
-		pl.EntryCreditBlock, err = entryCreditBlock.NextECBlock(nil)
+		if pl.DBHeight > 0 {
+			pl.DirectoryBlock, _ = state.GetDB().FetchDBlockByHeight(pl.DBHeight)
+			pl.AdminBlock, _ = state.GetDB().FetchABlockByHeight(pl.DBHeight)
+			pl.EntryCreditBlock, _ = state.GetDB().FetchECBlockByHeight(pl.DBHeight)
+		} else {
+			pl.DirectoryBlock = directoryBlock.NewDirectoryBlock(nil)
+			pl.AdminBlock = adminBlock.NewAdminBlock(nil)
+			pl.EntryCreditBlock, err = entryCreditBlock.NextECBlock(nil)
+		}
```

Comment:
it seems like maybe I should add a change like this
```
diff --git a/state/processList.go b/state/processList.go
index da9866957..73e3f471f 100644
--- a/state/processList.go
+++ b/state/processList.go
@@ -1167,9 +1167,13 @@ func NewProcessList(state interfaces.IState, previous *ProcessList, dbheight uin
                pl.EntryCreditBlock, err = entryCreditBlock.NextECBlock(previous.EntryCreditBlock)
        } else {
                if pl.DBHeight > 0 {
-                       pl.DirectoryBlock, _ = state.GetDB().FetchDBlockByHeight(pl.DBHeight)
-                       pl.AdminBlock, _ = state.GetDB().FetchABlockByHeight(pl.DBHeight)
-                       pl.EntryCreditBlock, _ = state.GetDB().FetchECBlockByHeight(pl.DBHeight)
+                       prevDBlock, _ := state.GetDB().FetchDBlockByHeight(pl.DBHeight-1)
+                       prevAblock, _ := state.GetDB().FetchABlockByHeight(pl.DBHeight-1)
+                       prevECBlock, _ := state.GetDB().FetchECBlockByHeight(pl.DBHeight-1)
+
+                       pl.DirectoryBlock = directoryBlock.NewDirectoryBlock(prevDBlock)
+                       pl.AdminBlock = adminBlock.NewAdminBlock(prevAblock)
+                       pl.EntryCreditBlock, err = entryCreditBlock.NextECBlock(prevECBlock)
                } else {
                        pl.DirectoryBlock = directoryBlock.NewDirectoryBlock(nil)
                        pl.AdminBlock = adminBlock.NewAdminBlock(nil)

```
My worry is that these blocks have already been stored - whereas generally Process Lists seem to only hold un-commited db blocks